File: Captain.Plugins.BuiltIn/Encoders/PNGCaptureEncoder.cpp
#include "PNGCaptureEncoder.h"

using namespace System::Drawing::Imaging;

namespace Captain {
  namespace Plugins {
    namespace BuiltIn {
      void PNGCaptureEncoder::Encode(Bitmap^ bmp, Stream^ outStream) {
        reinterpret_cast<Image^>(bmp)->Save(outStream, ImageFormat::Png);
      }
    }
  }
}

File: Captain.Plugins.BuiltIn/Source/Encoders/PngCaptureEncoder.cs
using System.Drawing;
using System.Drawing.Imaging;
using System.IO;
using Captain.Common;

namespace Captain.Plugins.BuiltIn {
  /// <summary>
  ///   Encodes still captures as PNG images
  /// </summary>
  [DisplayName("PNG")]
  public class PngCaptureEncoder : IStaticEncoder {
    /// <summary>
    ///   Encoder information
    /// </summary>
    public EncoderInfo EncoderInfo => new EncoderInfo() {
      Extension = "png",
      MediaType = "image/png"
    };

    /// <summary>
    ///   Performs the encoding
    /// </summary>
    /// <param name="bitmap">Capture</param>
    /// <param name="outputStream">Output stream</param>
    public void Encode(Bitmap bitmap, Stream outputStream) {
      bitmap.Save(outputStream, ImageFormat.Png);
    }
  }
}
